src/main/java/org/apache/cassandra/sidecar/routes/ListSnapshotFilesHandler.java

```diff
@@ -0,0 +1,197 @@
+                  {
+                      context.fail(new HttpException(HttpResponseStatus.NOT_FOUND.code(), cause.getMessage()));
+                  }
+                  else
+                  {
+                      context.fail(new HttpException(HttpResponseStatus.BAD_REQUEST.code(), "Invalid request for "
+                                                                                            + requestParams));
+                  }
+              });
+    }
+
+    private ListSnapshotFilesResponse buildResponse(String host,
+                                                    ListSnapshotFilesRequest request,
+                                                    List<Pair<String, FileProps>> fileList)
+    {
+        InstanceMetadata instanceMetadata = instancesConfig.instanceFromHost(host);
+        int sidecarPort = instanceMetadata.port();
+        Path dataDirPath = Paths.get(instanceMetadata.dataDirs().get(DATA_DIR_INDEX));
+        ListSnapshotFilesResponse response = new ListSnapshotFilesResponse();
+        String snapshotName = request.getSnapshotName();
+
+        for (Pair<String, FileProps> file : fileList)
+        {
+            Path pathFromDataDir = dataDirPath.relativize(Paths.get(file.getLeft()));
+
+            String keyspace = request.getKeyspace() != null
+                              ? request.getKeyspace()
+                              : pathFromDataDir.getName(0).toString();
+            // table name might include a dash (-) with the table UUID so we always use it as part of the response
+            String tableName = pathFromDataDir.getName(1).toString();
+            String fileName = pathFromDataDir.subpath(4, pathFromDataDir.getNameCount()).toString();
+
+            response.addSnapshotFile(new ListSnapshotFilesResponse.FileInfo(file.getRight().size(),
+                                                                            host,
+                                                                            sidecarPort,
+                                                                            DATA_DIR_INDEX,
+                                                                            snapshotName,
+                                                                            keyspace,
+                                                                            tableName,
+                                                                            fileName));
+        }
+        return response;
+    }
+
+    private ListSnapshotFilesRequest extractParamsOrThrow(final RoutingContext context)
+    {
+        boolean includeSecondaryIndexFiles =
+        "true".equalsIgnoreCase(context.request().getParam(INCLUDE_SECONDARY_INDEX_FILES, "false"));
+
+        return new ListSnapshotFilesRequest(context.pathParam("keyspace"),
+                                            context.pathParam("table"),
+                                            context.pathParam("snapshot"),
```

Comment:
weak nit: those url segment names can be constants. And the handler can define the routes to be referenced in the main module.